src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -4653,6 +4680,75 @@ class FilegroupClause(BaseSegment):
     )
 
 
+class CreateFullTextCatalogStatementSegment(BaseSegment):
+    """CREATE FULLTEXT CATALOG statement segment.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/create-fulltext-catalog-transact-sql
+    """
+
+    type = "create_fulltext_catalog_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        "FULLTEXT",
+        "CATALOG",
+        Ref("ObjectReferenceSegment"),
+        AnyNumberOf(
```

Comment:
Why `AnyNumberOf`? The syntax shows these are all optional, but can be used only **once**, so I'd go with an `Optional=True` for each sequence